resolver.go

```diff
@@ -65,34 +67,45 @@ func (c *resolver) ResolveNow(grpcresolver.ResolveNowOptions) {
 		} else {
 			c.logger.Errorln(err.Error())
 		}
-		c.cc.ReportError(err)
-		return
+		return nil, err
 	}
 
 	addrs := make([]grpcresolver.Address, len(output.Instances))
 	for i, instance := range output.Instances {
 		addrs[i] = httpInstanceSummaryToAddr(instance)
 	}
 
-	c.cc.UpdateState(grpcresolver.State{Addresses: addrs})
+	return &grpcresolver.State{Addresses: addrs}, nil
 }
 
 func (c *resolver) Close() {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	if c.isClosed {
-		return
-	}
-
-	c.isClosed = true
+	c.cancel()
 	c.ticker.Stop()
+	c.wg.Wait()
 }
 
-func (c *resolver) watch() {
+func (c *resolver) watcher() {
```

Comment:
Ditto

Reply:
I referred this : https://github.com/grpc/grpc-go/blob/master/internal/resolver/dns/dns_resolver.go#L214